osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -146,54 +155,135 @@ public Storage LocateStableStorage()
                 Logger.Error(e, "Stable installation could not be found; disabling file based IPC");
             }
 
-            return Storage;
+            return IPCStorage;
         }
 
+        public static bool CheckExists(string p) => File.Exists(Path.Combine(p, "ipc.txt"));
```

Comment:
I'd rename to something like `IPCFileExistsInDirectory`. This method should ideally not be `public`, too.